eth/api.go

```diff
@@ -409,11 +409,18 @@ type storageEntry struct {
 }
 
 // StorageRangeAt returns the storage at the given block height and transaction index.
-func (api *DebugAPI) StorageRangeAt(ctx context.Context, blockHash common.Hash, txIndex int, contractAddress common.Address, keyStart hexutil.Bytes, maxResult int) (StorageRangeResult, error) {
-	// Retrieve the block
-	block := api.eth.blockchain.GetBlockByHash(blockHash)
+func (api *DebugAPI) StorageRangeAt(ctx context.Context, blockNrOrHash rpc.BlockNumberOrHash, txIndex int, contractAddress common.Address, keyStart hexutil.Bytes, maxResult int) (StorageRangeResult, error) {
+	var block *types.Block
+	if number, ok := blockNrOrHash.Number(); ok {
+		block = api.eth.blockchain.GetBlockByNumber(uint64(number))
+	} else if hash, ok := blockNrOrHash.Hash(); ok {
+		block = api.eth.blockchain.GetBlockByHash(hash)
+	} else {
+		return StorageRangeResult{}, errors.New("either block number or block hash must be specified")
+	}
+
 	if block == nil {
-		return StorageRangeResult{}, fmt.Errorf("block %#x not found", blockHash)
+		return StorageRangeResult{}, fmt.Errorf("block %d not found", block)
```

Comment:
You cannot `%d` a `*types.Block`. And that `*types.Block` is also `nil` here. Maybe you meant to do something like this? 

```suggestion
		return StorageRangeResult{}, fmt.Errorf("block %v not found", blockNrOrHash)
```